apps/server/src/modules/pseudonym/controller/api-test/pseudonym.api.spec.ts

```diff
@@ -0,0 +1,158 @@
+				const { studentUser, studentAccount } = UserAndAccountTestFactory.buildStudent({ school }, []);
+				const pseudonymString: string = new ObjectId().toHexString();
+				const externalToolEntity: ExternalToolEntity = externalToolEntityFactory.buildWithId();
+				const pseudonym: ExternalToolPseudonymEntity = externalToolPseudonymEntityFactory.buildWithId({
+					pseudonym: pseudonymString,
+					toolId: externalToolEntity.id,
+					userId: studentUser.id,
+				});
+
+				await em.persistAndFlush([studentAccount, studentUser, pseudonym, externalToolEntity, school]);
+				em.clear();
+
+				const loggedInClient: TestApiClient = await testApiClient.login(studentAccount);
+
+				return { loggedInClient, pseudonym, pseudonymString };
+			};
+
+			it('should return a pseudonymResponse', async () => {
+				const { loggedInClient, pseudonymString, pseudonym } = await setup();
+
+				const response: Response = await loggedInClient.get(`${pseudonymString}`);
+
+				expect(response.status).toEqual(HttpStatus.OK);
+				expect(response.body).toEqual<PseudonymResponse>(
+					expect.objectContaining<Partial<PseudonymResponse>>({
+						id: pseudonym.id,
+						userId: pseudonym.userId.toString(),
+						toolId: pseudonym.toolId.toString(),
+					}) as PseudonymResponse
+				);
+			});
+		});
+
+		describe('when pseudonym is not connected to the users school', () => {
+			const setup = async () => {
+				const school: School = schoolFactory.buildWithId();
+				const { studentUser, studentAccount } = UserAndAccountTestFactory.buildStudent();
+				const { teacherUser, teacherAccount } = UserAndAccountTestFactory.buildTeacher({ school });
+				const pseudonymString: string = new ObjectId().toHexString();
+				const externalToolEntity: ExternalToolEntity = externalToolEntityFactory.buildWithId();
+				const pseudonym: ExternalToolPseudonymEntity = externalToolPseudonymEntityFactory.buildWithId({
+					pseudonym: pseudonymString,
+					toolId: externalToolEntity.id,
+					userId: teacherUser.id,
+				});
+
+				await em.persistAndFlush([
+					studentAccount,
+					studentUser,
+					teacherUser,
+					teacherAccount,
+					pseudonym,
+					externalToolEntity,
+					school,
+				]);
+				em.clear();
+
+				const loggedInClient: TestApiClient = await testApiClient.login(studentAccount);
+
+				return { loggedInClient, pseudonymString };
+			};
+
+			it('should return forbidden', async () => {
+				const { loggedInClient, pseudonymString } = await setup();
+
+				const response: Response = await loggedInClient.get(`${pseudonymString}`);
+
+				expect(response.status).toEqual(HttpStatus.FORBIDDEN);
+			});
+		});
+
+		describe('when pseudonym does not exist in db', () => {
+			const setup = async () => {
+				const school: School = schoolFactory.buildWithId();
+				const { studentUser, studentAccount } = UserAndAccountTestFactory.buildStudent({ school });
+				const pseudonymString: string = new ObjectId().toHexString();
```

Comment:
idk if we should use ObjectIds as mock data for pseudonyms since it suggests that pseudonyms are ObjectIds but they are actually uuidv4s